cpp/src/ffi_go/ir/decoder.h

```diff
@@ -0,0 +1,95 @@
+#ifndef FFI_GO_IR_DECODER_H
+#define FFI_GO_IR_DECODER_H
+// header must support C, making modernize checks inapplicable
+// NOLINTBEGIN(modernize-deprecated-headers)
+// NOLINTBEGIN(modernize-use-trailing-return-type)
+// NOLINTBEGIN(modernize-use-using)
+
+#ifdef __cplusplus
```

Comment:
`extern "C"` block should not wrap header includes. Same for the rest of header files